### PR TITLE
Update to new api to set static edp xml content [11002]

### PR DESCRIPTION
**Pull request by richiware**
Depends on eProsima/Fast-DDS#1818

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/824/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/825/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/826/)

File: docs/fastdds/discovery/static.rst [hunk 1]
```diff
@@ -217,22 +217,33 @@ This is done by setting the user ID on the |DataReaderQoS|/|DataWriterQoS|:
...
-+----------------------------------------------------------------------------------------------------------------------+
-
+On the local DomainParticipant, you can load STATIC EDP configuration content specifying the file containing it.
+
++------------------------------------------------------+
+| **C++**                                              |
++------------------------------------------------------+
+| .. literalinclude:: /../code/DDSCodeTester.cpp       |
+|    :language: c++                                    |
+|    :start-after: //CONF_STATIC_DISCOVERY_XML_FILE    |
```
Comment (EduPonz) [inline docs/fastdds/discovery/static.rst:227]: This is not correct

File: docs/fastdds/discovery/static.rst [hunk 2]
```diff
@@ -217,22 +217,33 @@ This is done by setting the user ID on the |DataReaderQoS|/|DataWriterQoS|:
...
++------------------------------------------------------+
+
+Or you can specify the STATIC EDP configuration content directly.
+
++------------------------------------------------------+
+| **C++**                                              |
++------------------------------------------------------+
+| .. literalinclude:: /../code/DDSCodeTester.cpp       |
+|    :language: c++                                    |
+|    :start-after: //CONF_STATIC_DISCOVERY_XML_DATA    |
```
Comment (EduPonz) [inline docs/fastdds/discovery/static.rst:246]: This is not correct

File: docs/fastdds/discovery/static.rst [hunk 3]
```diff
@@ -55,8 +55,8 @@ This can be done either by code or using an XML configuration file:
 
 .. _`static_xml`:
 
-STATIC EDP XML Files Specification
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+STATIC EDP XML Content Specification
```
Comment (EduPonz) [inline docs/fastdds/discovery/static.rst:58]: As I said in eProsima/Fast-DDS#1818, I like configuration better
```suggestion
STATIC EDP XML Configuration Specification
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: code/DDSCodeTester.cpp [hunk 4]
```diff
@@ -734,13 +739,30 @@ void dds_discovery_examples()
         //!--
     }
     {
-        //CONF_STATIC_DISCOVERY_XML
+        //CONF_STATIC_DISCOVERY_XML_FILE
```
Comment (EduPonz) [inline code/DDSCodeTester.cpp:742]: I think this one should be removed in favor of the other one

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]

**Comment by EduPonz** [comment]
Please correct DCO

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/829/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/830/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/831/)

**Review by EduPonz — APPROVED** [review]

**Pull request merged**